### PR TITLE
Send burnt tx fees to burn contract to finish up EIP-1559 fee handling

### PR DESCRIPTION
**Pull request by roberto-bayardo**
This PR implements the accounting for Polygon fee burning as described in https://forum.polygon.technology/t/eip1559-implementation-on-mumbai-testnet/399.  Base fees are to be sent to the appropriate burn contract address which this PR allows to be specified in the client config.

### CODE REVIEW

File: polygon/client.go [hunk 1]
```diff
@@ -1500,3 +1539,22 @@ func toCallArg(msg ethereum.CallMsg) interface{} {
 	}
 	return arg
 }
+
+// This implementation is taken from github.com/maticnetwork/bor/params/config.go
+//
+// TODO: Depend on maticnetwork fork of go-ethereum instead of stock geth
+func (ec *Client) CalculateBurntContract(number uint64) string {
```
Comment (yuga-cb) [inline polygon/client.go:1546]: Nit: change number -> blockNumber to make it more descriptive?

File: polygon/client.go [hunk 2]
```diff
@@ -1500,3 +1539,22 @@ func toCallArg(msg ethereum.CallMsg) interface{} {
 	}
 	return arg
 }
+
+// This implementation is taken from github.com/maticnetwork/bor/params/config.go
```
Comment (yuga-cb) [inline polygon/client.go:1543]: Not seeing it here: https://github.com/maticnetwork/bor/blob/master/params/config.go

Comment (roberto-bayardo) [reply polygon/client.go:1543]: Yeah it's not in master (yet). You can see it here: https://github.com/maticnetwork/bor/blob/c227a072418626dd758ceabffd2ea7dadac6eecb/params/config.go#L527

File: polygon/client.go (hunk 1 above)
Comment (roberto-bayardo) [reply polygon/client.go]: changed to blockNum

File: polygon/client.go (hunk 2 above)
Comment (roberto-bayardo) [reply polygon/client.go]: updated the comment with this permalink

**Review by yuga-cb — COMMENTED** [review]
LGTM!

**Review by lorines — APPROVED** [review]

**Pull request merged**